lbrynet/lbrynet_daemon/Daemon.py

```diff
@@ -1605,25 +1610,91 @@ def jsonrpc_stream_cost_estimate(self, name, size=None):
-            'file_path' (optional): str, path to file to be associated with name, if not given
-                                    the stream from your existing claim for the name will be used
-            'fee' (optional): dict, FeeValidator compliant (i.e. {'LBC':{'amount':10}} )
+            'file_path' (optional): str, path to file to be associated with name, if provided
+                                    a lbry stream of this file will be used in 'sources'.
+                                    If no path is given but a metadata dict is provided the source
+                                    from the given metadata will be used.
+            'fee' (optional): dict, {currency_symbol: {'amount': float, 'address': str, optional}}
+                              supported currencies: LBC, USD, BTC
+                              If an address is not provided a new one will be generated
+            'title'(optional): str
+            'description'(optional): str
+            'author'(optional): str
+            'language'(optional): str, language code
+            'license'(optional): str
+            'license_url'(optional): str
+            'thumbnail'(optional): str
+            'preview'(optional): str
+            'nsfw'(optional): bool
+            'sources'(optional): dict
+
         Returns:
-            'tx' : hex encoded transaction
-            'txid' : txid of resulting transaction
-            'nout' : nout of the resulting support claim
-            'fee' : fee paid for the claim transaction
+            'success' : True if claim was successful , False otherwise
+            'reason' : if not successful, give reason
+            'txid' : txid of resulting transaction if succesful
+            'nout' : nout of the resulting support claim if successful
+            'fee' : fee paid for the claim transaction if successful
             'claim_id' : claim id of the resulting transaction
         """
 
+        verify_name_characters(name)
+
+        if bid <= 0.0:
+            raise Exception("Invalid bid")
+
+        metadata = metadata or {}
+        if fee is not None:
+            metadata['fee'] = fee
+        if title is not None:
+            metadata['title'] = title
+        if description is not None:
+            metadata['description'] = description
+        if author is not None:
+            metadata['author'] = author
+        if language is not None:
+            metadata['language'] = language
+        if license is not None:
+            metadata['license'] = license
+        if license_url is not None:
+            metadata['license_url'] = license_url
+        if thumbnail is not None:
+            metadata['thumbnail'] = thumbnail
+        if preview is not None:
+            metadata['preview'] = preview
+        if nsfw is not None:
+            metadata['nsfw'] = nsfw
+        if sources is not None:
+            metadata['sources'] = sources
+        if fee is not None:
```

Comment:
if fee is specified in metadata instead of argument,the address will never be added
I fixed this in:
https://github.com/lbryio/lbry/pull/531